docs/translating.md

```diff
@@ -0,0 +1,169 @@
+# UI translation
+
+The website UI is automatically translated in [available languages](TODO: link
+to translation.ts). This is achieved with
+[`react-i18next`](https://react.i18next.com/).
+
+---
+
+<!-- vim-markdown-toc GitLab -->
+
+* [Workflows](#workflows)
+    * [Editing the source code](#editing-the-source-code)
+    * [Adding a new language](#adding-a-new-language)
+* [Architecture](#architecture)
+* [Available scripts](#available-scripts)
+    * [`generate-ui.js`](#generate-uijs)
+    * [`translate-ui.js`](#translate-uijs)
+    * [`translate-md.js`](#translate-mdjs)
+    * [`translate-faq.js`](#translate-faqjs)
+    * [`translate-releases.js`](#translate-releasesjs)
+
+<!-- vim-markdown-toc -->
+
+---
+
+## Workflows
+
+### Editing the source code
+
+When adding a new string containing a content which need to be translated,
+you need to wrap it inside a [`Trans`](https://react.i18next.com/latest/trans-component) React component
+or inside a function call to [`t`](https://react.i18next.com/latest/usetranslation-hook).
+
+Once the content modified,
+
+1. you can update the resource files with the [`generate-ui.js`](#generate-ui.js) script.
+    <details>
+    <summary>Script output example</summary>
+
+    ```
+    > npm run generate:ui
+
+    > nosgestesclimat-site@2.1.0 generate:ui
+    > node scripts/i18n/generate-ui.js
+
+    Static analysis of the source code...
+    Adding missing entries...
+    + Added 2 translations:
+        Cacher l'objectif
+        catégorie complétée
+    ~ Updated 1 translations:
+        components.stats.StatsContent.enSavoirPlus
+    - Missing 2 translations:
+        components.conversation.select.NumberedMosaic.choixAFaire
+        components.conversation.select.NumberedMosaic.choixEnTrop
+    Writting resources in /home/emile/Projects/datagir/nosgestesclimat-site/source/locales/ui/ui-fr.json...
+
+    ```
+    </details>
+
+    For _added_ and _updated_ entries, their translation are automatically
```

Comment:
> For _added_ and _updated_ entries, their translation are automatically
    retrieved from the source code
    
   What does it mean ? You mean that the script checks if the same texte has already been translated elsewhere ?

Reply:
It means that the value is provided in the source code for example from a `<Trans>` component containing the text to translate:

```typescript
<Trans>Mon texte à traduire</Trans>
```

here, we can automatically retrieve and update the value.
However, when only a key is provided like in `t('i18n.key.id')`, the value will be `NO_TRANSLATION` and the user will need to add it manually.